source/lac/trilinos_sparsity_pattern.cc

```diff
@@ -437,6 +438,7 @@ namespace TrilinosWrappers
                std_cxx11::shared_ptr<Epetra_FECrsGraph> &graph,
                std_cxx11::shared_ptr<Epetra_CrsGraph>   &nonlocal_graph)
     {
+      nonlocal_graph.reset ();
```

Comment:
Good catch. I'm not sure if it worked before I reshuffled the code into separate functions, but right now it's untested. I'll construct a test case during the next days. It is somewhat of a corner because you first need to create a sparsity that is set up with writeable rows and later create one where off-processor rows are filled on demand; it is not that likely in user code but now that you've spotted the case it's good to also test it. Thanks.